toolsrc/src/tls12-download.c

```diff
@@ -0,0 +1,263 @@
+#include <Windows.h>
+#include <process.h>
+#include <winhttp.h>
+/*
+ * This program must be as small as possible, because it is committed in binary form to the
+ * vcpkg github repo to enable downloading the main vcpkg program on Windows 7, where TLS 1.2 is
+ * unavailable to PowerShell.
+ * To that end it avoids using C runtime functions (beyond the vcruntime ones the compiler
+ * injects itself).
+ * (In testing as of 2021-01-07, this version that doesn't link with the CRT is ~8kb, whereas a
+ * hello world program that does link with the CRT is ~300kb)
+ */
+
+void __declspec(noreturn) win32_abort() { TerminateProcess(GetCurrentProcess(), 3); }

```

Comment:
```suggestion
void __declspec(noreturn) win32_abort()
{
    TerminateProcess(GetCurrentProcess(), 3);
    for (;;) {}
}
```

`TerminateProcess` is asynchronous; thus, more likely than not, this function will fall off the end. (Nicole notes that this is C, and thus that infinite loops are defined behavior. If you'd like More Defined Behavior, you can add an access to a volatile.)

Reply:
https://github.com/MicrosoftDocs/sdk-api/pull/626

Reply:
Added comment linking to that PR.